server/common/annotations/hosted_tiledb.py

```diff
@@ -6,21 +6,31 @@
 import pandas as pd
 import tiledb
 from flask import current_app
+from tiledb.libtiledb import TileDBError
 
 from server.common.annotations.annotations import Annotations
 from server.common.errors import AnnotationCategoryNameError
 from server.common.utils.sanitization_utils import sanitize_values_in_list
 from server.common.utils.type_conversion_utils import get_dtypes_and_schemas_of_dataframe, get_dtype_of_array
-from server.db.cellxgene_orm import CellxGeneDataset, Annotation
+from server.db.cellxgene_orm import Annotation
 
 
 class AnnotationsHostedTileDB(Annotations):
     CXG_ANNO_COLLECTION = "cxg_anno_collection"
 
-    def __init__(self, directory_path, db):
+    def __init__(self, directory_path, db, server_config):
         super().__init__()
         self.db = db
-        self.directory_path = directory_path
+        if directory_path[-1] == "/":
+            self.directory_path = directory_path
+        else:
+            self.directory_path = directory_path + "/"
+        tiledb_config = server_config.adaptor__cxg_adaptor__tiledb_ctx
+        self.ctx = tiledb.Ctx(config=tiledb_config)
+        try:
+            tiledb.default_ctx(tiledb_config)
+        except TileDBError:
+            pass
```

Comment:
What I meant earlier is you can remove all of these lines.   
We've already created a tiledb_config in the CxgAdaptor class.  
The fix was to  add "tiledb.default_ctx( CxgAdaptor.tiledb_ctx) to the "set_tiledb_context" function in CxgAdaptor.
Once that is done, there is no longer a need to pass in a ctx into any of the tiledb functions.
So, then you can remove self.ctx here, and the places where you used it.

Reply:
sorry I misunderstood your earlier message, just updated